File: Algorithm/C++/1000+/1029. Two City Scheduling.cc
/*
 * Created on Sun May 05 2019 15:48:59
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

// sort with custom function
#include <vector>

using std::vector;

class Solution {
 public:
  int twoCitySchedCost(vector<vector<int>>& costs) {
    std::sort(begin(costs), end(costs), [](vector<int> &v1, vector<int> &v2) {
      return (v1[0] - v1[1] < v2[0] - v2[1]);
    });
    int result = 0;
    for (int i = 0; i < (costs.size() >> 1); ++i) {
      result += costs[i][0] + costs[i + （cs.size() >> 1)][1];
    }
    return result;
  }
};

File: Algorithm/Go/1000+/1029. Two City Scheduling.go
/*
 * Created on Sun May 05 2019 16:52:21
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

 // sort with custom compare function
import "sort"

func twoCitySchedCost(costs [][]int) int {
	sort.SliceStable(costs, func(i, j int) bool {
		return costs[i][0] - costs[i][1] < costs[j][0] - costs[j][1]
	})
	result, half := 0, len(costs)>>1
	for i := 0; i < half; i++ {
		result += costs[i][0] + costs[i+half][1]
	}
	return result
}